packages/block-library/src/preformatted/edit.native.js

```diff
@@ -29,7 +29,8 @@ export default function PreformattedEdit( props ) {
 		...( style?.color && { color: style.color } ),
 	};
 	const hasBaseColors =
-		style?.baseColors && Object.entries( style.baseColors ).length !== 0;
+		style?.baseColors?.color &&
+		Object.entries( style.baseColors ).length !== 0;
```

Comment:
Actually, I need to update this code, it should only keep the `style?.baseColors?.color` check, before it would pass an empty object if no global styles were available, but now it'll pass an object with the [expected structure](https://github.com/WordPress/gutenberg/pull/50108/files#diff-7ace6cc2458af47bd9067d7968d1e8f6b724b3786690ae9d133a65aeea5d00ddR31) (although values can be undefined). I'll update it 👍

Reply:
Updated in https://github.com/WordPress/gutenberg/pull/50108/commits/98eada3d0d4866febe9744b7772087490d377dce